OTRExporter/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR)

```

Comment:
Running `cmake -S . -B build-cmake -DCMAKE_TOOLCHAIN_FILE=../soh/CMake/Mac-gnu-toolchain.cmake` for OTRExporter and getting:
```
Could NOT find PNG (missing: PNG_LIBRARY PNG_PNG_INCLUDE_DIR)
```

which is weird given its installed via brew.. any hints?

Reply:
does it work without the toolchain file?

Reply:
oh it does! but then `cmake --build build-cmake ` fails because it defaults to clang - when i want it to use gcc

Reply:
No toolchain file means that the defaults are used. If you set these values in the toolchain file, it should work as well...

Reply:
Is there a benefit to using the toolchain rather than specifying these two flags?

Reply:
ok.. now I tried to run:
`cmake -S . -B build-cmake -DCMAKE_C_COMPILER=gcc-12 -DCMAKE_CXX_COMPILER=g++-12 -DCMAKE_SYSTEM_NAME=Darwin`

and that failed.. seems like the problem is the `DCMAKE_SYSTEM_NAME`